packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3459,13 +3610,32 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     return null;
   }
 
+  @override
+  T dependOnInheritedWidgetOfExactType<T extends InheritedWidget>({Object aspect}) {
+    assert(_debugCheckStateIsActiveForAncestorLookup());
+    final InheritedElement ancestor = _inheritedWidgets == null ? null : _inheritedWidgets[T];
+    if (ancestor != null) {
+      assert(ancestor is InheritedElement);
+      return inheritFromElement(ancestor, aspect: aspect);
+    }
+    _hadUnsatisfiedDependencies = true;
+    return null;
+  }
+
   @override
   InheritedElement ancestorInheritedElementForWidgetOfExactType(Type targetType) {
```

Comment:
Can we implement the old API by calling the new API to avoid duplication?

Reply:
Unfortunately not in this direction (the new can be implemented with the old) because we cannot extract `T` from `Type`.